proxy/ptunnel/commands/agent.go

```diff
@@ -4,47 +4,81 @@ import (
 	"fmt"
 	"github.com/enfabrica/enkit/lib/client"
 	"github.com/enfabrica/enkit/lib/kcerts"
+	"github.com/enfabrica/enkit/lib/kflags"
 	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"strings"
 )
 
 type agentConfig struct {
-	Print          bool
-	ListIdentities bool
+	Print         bool
+	RemainingArgs []string
 }
 
 func NewAgentCommand(bf *client.BaseFlags) *cobra.Command {
 	agentConfig := &agentConfig{}
 	c := &cobra.Command{
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return RunAgentCommand(bf, agentConfig)
-		},
-		Use:   "agent",
-		Short: "commands for the enkit specific ssh-agent",
+		Use:   "agent [SubCommands] -- [Command]",
+		Short: "commands for the enkit specific ssh-agent, anything passed in will execute with SSH_AUTH_SOCK and SSH_AGENT_PID set for the enkti agent.",
 	}
 	// Note the following is intended to be user friendly, identities here are cert principals
-	c.Flags().BoolVarP(&agentConfig.ListIdentities, "list-identities", "l", false, "list the identities loaded current in the agent")
 	c.Flags().BoolVarP(&agentConfig.Print, "print", "p", false, "print the socket and PID of the running agent")
+	c.AddCommand(NewRunAgentCommand(c, bf, agentConfig))
+	c.AddCommand(NewListAgentCommand(bf))
+	return c
+}
+func NewListAgentCommand(bf *client.BaseFlags) *cobra.Command {
+	c := &cobra.Command{
+		Use: "list",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			agent, err := kcerts.FindSSHAgent(bf.Local, bf.Log)
+			if err != nil {
+				return err
+			}
+			principals, err := agent.Principals()
+			if err != nil {
+				return err
+			}
+			for _, p := range principals {
+				fmt.Printf("PKS: %s Identities: %v ValidFor: %s \n", p.MD5, p.Principals, p.ValidFor.String())
+			}
+			return nil
+		},
+	}
+	return c
+}
+
+func NewRunAgentCommand(parent *cobra.Command, bf *client.BaseFlags, config *agentConfig) *cobra.Command {
+	c := &cobra.Command{
+		Use:   "run -- [COMMAND]",
+		Short: "Runs the following command using the enkit ssh-agent",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunAgentCommand(parent, bf, config, args)
+		},
+	}
 	return c
 }
 
-func RunAgentCommand(bf *client.BaseFlags, config *agentConfig) error {
+func RunAgentCommand(command *cobra.Command, bf *client.BaseFlags, config *agentConfig, args []string) error {
 	agent, err := kcerts.FindSSHAgent(bf.Local, bf.Log)
 	if err != nil {
 		return err
 	}
 	if config.Print {
```

Comment:
Would move this to a different command, as now it is impossible to just print the values and exit? as even an empty args is a valid command. eg, `print` or `config`, so eg one can manually do:
```
eval `enkit agent print`
```
to configure the same agent as enkit.